netbox/dcim/models/device_components.py

```diff
@@ -904,6 +904,12 @@ def clean(self):
                 ).format(untagged_vlan=self.untagged_vlan)
             })
 
+        # Validate that tagged-all payload does not include tagged_vlans
+        if self.mode != InterfaceModeChoices.MODE_TAGGED and self.tagged_vlans:
```

Comment:
I think this needs to go on the serializer, as we can't check against many-to-many assignments under the model's `clean()` method. (We should ensure that it applies for VM interfaces too.)